oat-sa/expr-eval
# Silentmatt v1.2.3

### Related to: https://oat-sa.atlassian.net/browse/ADF-1166

### Summary

Add changes from version `1.2.3` of the base repository.

Also, update the dependencies: these are only tooling, so they should not harm.

### Details

Version `1.2.3` is the last version before the next major version. Moving to version `2` would need more checks and a possible refactoring of the components using it (aka the scientific calculator).
Besides the fixes coming from the base repo, security issues spotted by `dependabot` are also fixed by updating dependencies.

### How to test
- install the package, and see that no vulnerabilities are found: `npm i`
- run the unit tests: `npm test`
- install https://github.com/oat-sa/tao-core-ui-fe, then clone the pull request in place of ` node_modules/@oat-sa/expr-eval`:
    ```
    cd node_modules/@oat-sa/
    rm -rf expr-eval
    git clone git@github.com:oat-sa/expr-eval.git -b silentmatt-v1.2.3
    cd expr-eval
    npm i
    cd ../../..
    npm run build:all
    ```
    Then run the unit test (`npm run test:keepAlive`) and open the browser at http://127.0.0.1:5400/test/maths/calculator/scientificCalculator/test.html
